services/rsources/handler.go

```diff
@@ -108,9 +113,128 @@ func (*sourcesHandler) GetFailedRecords(_ context.Context, _ *sql.Tx, _ string,
 }
 
 func (sh *sourcesHandler) Delete(ctx context.Context, jobRunId string) error {
-	return sh.Extension.DropStats(ctx, jobRunId)
+	sqlStatement := `delete from "rsources_stats" where job_run_id = $1`
+	_, err := sh.localDB.ExecContext(ctx, sqlStatement, jobRunId)
+	return err
+}
+
+func (sh *sourcesHandler) DropStats(ctx context.Context, jobRunId string) error {
+	sqlStatement := `delete from "rsources_stats" where job_run_id = $1`
+	_, err := sh.localDB.ExecContext(ctx, sqlStatement, jobRunId)
+	return err
 }
 
 func (sh *sourcesHandler) CleanupLoop(ctx context.Context) error {
-	return sh.Extension.CleanupLoop(ctx)
+	err := sh.doCleanupTables(ctx)
+	if err != nil {
+		return err
+	}
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-time.After(config.GetDuration("RSOURCES_STATS_CLEANUP_INTERVAL", 24, time.Hour)):
```

Comment:
I find this line as doing too much, can we separate the configuration part from this logic?

Ideally, we can introduce a function that mocks this clock behaviour.

```
type sourcesHandler struct {
    cleanupTrigger func() <- chan
}
```

This function can get initialised based on the config, or be overwritten by custom code for testing